### PR TITLE
GitHub Actions-based CI

### PR DESCRIPTION
**Pull request by Exxion**
1. The free Travis CI plan is going away in under a month.
2. Travis has been preposterously slow recently.

This is a mostly-finished replacement for Travis. Initial setup by @DamianX, lots of changes by me. Just needs some finishing touches to bring it up to equivalence with Travis's current setup.
In addition to not having fucking hour-long queues, the actual time to perform the job is almost three times faster on this service, as well.
Once this is done we can drop Travis whenever we decide.

~~For some reason, one of the unit tests passes on Travis but fails on this. I think it's because the misc. persistence subsystem is not guaranteed to be initialized before roundstart. I do not know why it always seems to pass on Travis and always seems to fail on this, because it seems like it should just be completely random.~~

### CODE REVIEW

**Comment by PrimeDSS13** [comment]
sow the seeds of your own demise, tr * v * s.
image

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          key: BYOND-${{runner.os}}-${{env.BYOND_MAJOR}}.${{env.BYOND_MINOR}}
+
+      - name: ${{matrix.job-name}}
+        env:
+          DM_UNIT_TESTS: ${{matrix.dm-unit-tests}}
+        run: |
+          tools/travis/install-byond.sh
+          source $HOME/BYOND-${BYOND_MAJOR}.${BYOND_MINOR}/byond/bin/byondsetup
+          tools/travis/build.py 2>&1 | tee /dev/stderr | awk '/warning:|error:/ { exit 1 }'
+          tools/travis/run_tests.py 2>&1 | tee /dev/stderr | awk '/UNIT TEST FAIL/ { exit 1 }'
```
Comment (DamianX) [inline .github/workflows/ci.yml:82]: the reason I had sponge in there was because otherwise awk would immediately kill the process as soon as one test failed, and wouldn't show the rest of the failures. I think that will happen again without it.

Comment (Exxion) [reply .github/workflows/ci.yml:82]: It doesn't, but I can put it back in anyway if you can tell me how to install it in this system

Comment (DamianX) [reply .github/workflows/ci.yml:82]:
```
- name: Install moreutils
  run: sudo apt-get install moreutils
```
should work

Comment (DamianX) [reply .github/workflows/ci.yml:82]: By "it doesn't" you mean it doesn't stop at the first error?

Comment (Exxion) [reply .github/workflows/ci.yml:82]: Correct

Comment (DamianX) [reply .github/workflows/ci.yml:82]: then don't bother doing that

**Comment by Exxion** [comment]
That unit test passed Travis and failed this because Travis was setting up the server config and this wasn't, and for some reason this made the misc. persistence subsystem initialize faster
We should probably fix it so that regardless of config it's guaranteed to initialize before roundstart, but whatever

**Comment by Exxion** [comment]
Travis apparently notifies the IRC about builds. How much do we care about keeping that?

**Comment by DamianX** [comment]
-100%

**Comment by Exxion** [comment]
Well in that case this is ready. It *could* use more caching, but since the job that that would speed up is the one that already goes the fastest, I don't think it would make any difference at all in the total execution time.

**Comment by Exxion** [comment]
Oh right it's also missing the libvg tests but libvg is empty right now
I should probably add those anyway though

**Comment by Exxion** [comment]
I can do that later I guess

**Comment by DamianX** [comment]
libvg is dead, I think this is a good opportunity to start getting rid of it.
